MoneySplitter.Services/API/SearchApiService.cs

```diff
@@ -0,0 +1,33 @@
+﻿using MoneySplitter.Infrastructure;
+using MoneySplitter.Models;
+using MoneySplitter.Services.DataModels;
+using MoneySplitter.Services.Inerfaces;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace MoneySplitter.Services.API
+{
+    public class SearchApiService : ISearchApiService
+    {
+        private readonly IApiUrlBuilder _urlBuilder;
+        private readonly IQueryApiService _queryApiService;
+        private readonly IMapper _maper;
+
+        public SearchApiService(IApiUrlBuilder urlBuilder, IQueryApiService queryApiService, IMapper maper)
+        {
+            _urlBuilder = urlBuilder;
+            _queryApiService = queryApiService;
+            _maper = maper;
+        }
+
+        public async Task<IEnumerable<UserModel>> SearchUsersAsync(string query)
+        {
+            var searchUsersUri = _urlBuilder.SearchUsers(query);
+
+            var dataUsers = await _queryApiService.GetAsync<IEnumerable<DataUser>>(searchUsersUri);
```

Comment:
rename this to usersData or remove this variable, just make return await _query....